src/libertem/win_tweaks.py

```diff
@@ -47,10 +58,13 @@ def set_console_mode(mode, stream=sys.stdin):
 
 
 def disable_quickedit():
-    mode = get_console_mode()
-    mode &= ~ENABLE_QUICK_EDIT
-    mode |= ENABLE_EXTENDED_FLAGS
-    set_console_mode(mode)
+    try:
+        mode = get_console_mode()
+        mode &= ~ENABLE_QUICK_EDIT
+        mode |= ENABLE_EXTENDED_FLAGS
+        set_console_mode(mode)
+    except TypeError:
+        logger.info("In win_tweaks.py, get_console_mode() returns None.")
```

Comment:
I'm still not 100% happy with this. Catching the `WinError` in `get_console_mode` causes a `TypeError` in `disable_quickedit`, which then again has to be caught... why catch the `WinError` at all in `get_console_mode`, if you effectively convert it into a `TypeError` down here, which you then again have to catch?

About the logged message: this is currently a "statement": it just states something about code internals. Would a user know what to do with that? How about this instead:

"Quick Edit mode could not be disabled, you are probably using a console which doesn't support Quick Edit (like git bash)"

That way, if the user **knows** they are using a console which supports Quick Edit (like CMD), and get this message, they **know** something is wrong. If they are using git bash, they are not alarmed because the message tells them that this is to be expected.

Reply:
Are these things (like quick edit) a part of /related to win32 API? If not, where can I learn more about them? The Win 32 API documentation isn't much helpful.

Reply:
`GetConsoleMode` and `SetConsoleMode` are part of the Win32 API, yes. [This](https://docs.microsoft.com/en-us/windows/console/getconsolemode) is the API documentation for `GetConsoleMode`, which contains a short description of quick edit. From a user's perspective, [this is how you enable/disable it](https://tweaks.com/windows/36732/command-prompt-quick-edit-mode/) "manually".